specs/IsSmartScreenRequired.md

```diff
@@ -0,0 +1,100 @@
+
+It is much easier to indicate if individual WebView2s require SmartScreen than to have an end developer explicitly manage if SmartScreen should be enabled as a whole, especially when its different sets of WebView2s in different processes (like Excel's set of WebView2s and Word's set of WebView2s) all sharing the same user data folder.
+In this document we describe the new setting.
+
+
+# Description
+You can use CoreWebView2Settings.IsReputationCheckingRequired to control SmartScreen. SmartScreen is enabled or disabled per browser process, so all WebView2 applications sharing the same user data folder path also share SmartScreen being enabled or disabled.
+If CoreWebView2Setting.IsReputationCheckingRequired is true for any CoreWebView2 using the same user data folder, then SmartScreen is enabled. If CoreWebView2Setting.IsReputationCheckingRequired is false for all CoreWebView2 using the same user data folder, then SmartScreen is disabled.
+The default value for `IsReputationCheckingRequired` is true. When creating a new CoreWebVIew2, if it is not set CoreWebView2Settings.IsReputationCheckingRequired, the SmartScreen state of all CoreWebView2s using the same user data folder will be reset to true when the new CoreWebView2 is navigated or downloaded.
+
+Changes to `IsReputationCheckingRequired` take effect on the next navigation or download.
+
+If the option `--disable-features=msSmartScreenProtection` is specified when CoreWebView2Environment is created, then SmartScreen cannot be set through CoreWebView2Settings.IsReputationCheckingRequired. In this scenario, SmartScreen is always turned off.
+
+
+# Examples
+```cpp
+// member variable
+wil::com_ptr<ICoreWebView2Settings> m_webViewSettings;
+
+// isLocalContent is TRUE if the page is navigated to content that is completely
+// app-provided, with no user-provided content or web-served content.
+// Note that we must update the property before navigating to the content.
+void SettingsComponent::UpdateSmartScreenRequirementBeforeNavigating(bool isLocalContent)
+{
+   wil::com_ptr<ICoreWebView2Settings11> coreWebView2Settings11;
+    coreWebView2Settings11 =
+                m_webViewSettings.try_query<ICoreWebView2Settings11>();
+    if(coreWebView2Settings11) 
+    {
+        CHECK_FAILURE(coreWebView2Settings11->put_IsReputationCheckingRequired(!isLocalContent));
+    }
+}
+```
+
+```c#
+void ToggleSmartScreenRequired()
+{
+    var settings = webView2Control.CoreWebView2.Settings;
+    settings.IsReputationCheckingRequired = !settings.IsReputationCheckingRequired;
```

Comment:
The C# sample code should match the C++ sample code above

Reply:
updated.